[Flaky CI] Sometimes deploying prometheus artifacts results with runtime error: `invalid memory address or nil pointer dereference`

For example this prow [lane](https://prow.ci.kubevirt.io/view/gs/kubevirt-prow/pr-logs/pull/kubevirt_cluster-network-addons-operator/955/pull-e2e-cluster-network-addons-operator-nmstate-functests/1432628512748998656#1:build-log.txt%3A558) failing on CNAO on [this line](https://github.com/kubevirt/cluster-network-addons-operator/blob/main/cluster/up.sh#L26):
```
$(cluster::path)/cluster-up/up.sh
```

logs when fails looks like this:
```
prometheusrule.monitoring.coreos.com/kube-prometheus-rules created
prometheusrule.monitoring.coreos.com/prometheus-operator-rules created
prometheusrule.monitoring.coreos.com/prometheus-k8s-prometheus-rules created
service/prometheus-nodeport created
service/grafana-nodeport created
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x9ab01c]
goroutine 22 [running]:
kubevirt.io/kubevirtci/cluster-provision/gocli/cmd.scp.func2(0xc000196900)
	/home/prow/go/src/github.com/kubevirt/kubevirtci/cluster-provision/gocli/cmd/scp.go:191 +0x5c
created by kubevirt.io/kubevirtci/cluster-provision/gocli/cmd.scp
	/home/prow/go/src/github.com/kubevirt/kubevirtci/cluster-provision/gocli/cmd/scp.go:189 +0xabc
Makefile:130: recipe for target 'cluster-up' failed
make: *** [cluster-up] Error 2
+ EXIT_VALUE=2
+ set +o xtrace
```

--- a/cluster-provision/gocli/cmd/scp.go
+++ b/cluster-provision/gocli/cmd/scp.go
@@ -185,16 +185,18 @@ func scp(cmd *cobra.Command, args []string) error {
 		_, err = io.CopyN(target, stdout, int64(l))
 		errChan <- err
 	}()
+	wrPipe, err := session.StdinPipe()
+	if err != nil {
+		return fmt.Errorf("failed to open pipe: %v", err)
+	}
 
-	go func() {
-		wrPipe, _ := session.StdinPipe()
+	go func(wrPipe io.WriteCloser) {
 		defer wrPipe.Close()
-
 		fmt.Fprintf(wrPipe, "\x00")
 		fmt.Fprintf(wrPipe, "\x00")
 		fmt.Fprintf(wrPipe, "\x00")
 		fmt.Fprintf(wrPipe, "\x00")
-	}()
+	}(wrPipe)
 
 	err = session.Run("sudo -i /usr/bin/scp -qf " + src)
 

